src/project_creator/templates/project/{{cookiecutter.name}}/index.rst

```diff
@@ -1,6 +1,38 @@
-Table of Contents
-=================
+{% if build_config.component_info %}

```

Comment:
This index.rst is common for both variant and component reports. We use the build_config read from the "config.json" generated by CMake configure step.